reactjs/src/pages/rules.js

```diff
@@ -110,14 +113,55 @@ export default function LIST_RULES(props) {
         );
     }
 
+    //function to fetch total balance of rules
+    const fetchBalance = useCallback(async (balanceLink) => {
```

Comment:
@C-Carter10 balanceLink will always be undefined here, because on line 151, you're not passing anything to the fetchBalance call